README.md

```diff
@@ -337,131 +288,29 @@ apply plugin: "io.realm.kotlin"
-
-All platform differentiated implementations are kept in `platform`-packages with their current package hierarchy, to make it easier to keep track of the level of platform differentiation.
-
-## Writing Tests
-
-Currently all unit tests should be place in the `test/` project instead of `packages/library`. The reason for this is that we need to apply the Realm Compiler Plugin to the tests and this introduces a circular dependency if the tests are in `library`.
-
-Inside `tests/` there are 3 locations the files can be placed in:
-
-* `test/src/commonTest`
-* `test/src/androidTest`
-* `test/src/macosTest`
-
-Ideally all shared tests should be in `commonTest` with specific platform tests in `androidTest`/`macosTest`. However IntelliJ does not yet allow you to run common tests on Android from within the IDE](https://youtrack.jetbrains.com/issue/KT-46452), so we
-are using the following work-around:
-
-1) All "common" tests should be placed in the `test/src/androidtest/kotlin/io/realm/shared` folder. They should be written using only common API's. I'e. use Kotlin Test, not JUnit. This `io.realm.shared` package should only contain tests we plan to eventually move to `commonTest`.
-
-
-2) The `macosTest` shared tests would automatically be picked up from the `androidTests` as it is symlinked to `test/src/androidtest/kotlin/io/realm/shared`.
-
-
-3) This allows us to run and debug unit tests on both macOS and Android. It is easier getting the imports correctly using the macOS sourceset as the Android code will default to using JUnit.
- 
-
-All platform specific tests should be placed outside the `io.realm.shared` package, the default being `io.realm`.
-
-## Defining dependencies
+## Kotlin Memory Model and Coroutine compatibility
 
-All dependency versions and other constants we might want to share between projects are defined inside the file 
-`buildSrc/src/main/kotlin/Config.kt`. Any new dependencies should be added to this file as well, so we only have one
-location for these.
+Realm Kotlin is implemented against Kotlin's default memory model (the old one), but still supports running with the new memory model if enabled in the consuming project. See https://github.com/JetBrains/kotlin/blob/master/kotlin-native/NEW_MM.md#switch-to-the-new-mm for details on enabled the new memory model.
 
-## Debugging Kotlin/Native Tests
+By default Realm Kotlin depends and requires to run with Kotlin Coroutines version `1.6.0-native-mt`. To use Realm Kotlin with the non-`native-mt` version of Coroutines you will have to enable the new memory model and also disables our internal freezing to accomodate the new freeze transparency for Coroutine 1.6.0. See https://github.com/JetBrains/kotlin/blob/master/kotlin-native/NEW_MM.md#unexpected-object-freezing for more details on that.
```

Comment:
This is ambiguous 
> To use Realm Kotlin with the non-`native-mt` version of Coroutines you will have to enable the new memory model and also disables our internal freezing

how can a user diable our internal freezing to accommodate ...

Reply:
I updated the section a little bit to explicitly mention the Gradle options required + still link the JetBrains docs.